src/gitlab-loader.ts

```diff
@@ -15,13 +15,11 @@ export default class GitlabLoader {
 	}
 
 	loadIssues() {
+		// URL Encode the user filter settings
+		const filterItems = this.settings.filter.split("&").map(filterGroup => filterGroup.split('=').map((value, index) =>  index % 2 == 0 ? value : `${encodeURIComponent(value)}` ).join("="));
+		const apiURL = `${this.settings.gitlabApiUrl()}/issues?${filterItems.join('&')}`;
 
-		const urlParts: Array<string> = [
-			this.settings.gitlabUrl,
-			'/issues?' + this.settings.filter,
-		];
-
-		GitlabApi.load<Array<Issue>>(urlParts.join(''), this.settings.gitlabToken)
+		GitlabApi.load<Array<Issue>>(apiURL, this.settings.gitlabToken)
```

Comment:
This seems overly complicated. Would it not be better to construct the full URL first, and then pass the complete string to `encodeURI()` which is designed to URL encode a complete URL.

Reply:
You are right we can just run it through `encodeURI()`, I got a bit carried away.

I got hung up on the paragraph just below the examples in [Mozilla's documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#examples).

I'll change it to your suggestion 👍